internal/pkg/skuba/deployments/ssh/ssh.go

```diff
@@ -181,13 +192,19 @@ func (t *Target) internalSshWithStdin(silent bool, stdin string, command string,
 	return
 }
 
-func readerStreamer(reader io.Reader, outputChan chan<- string, description string, silent bool) {
+func readerStreamer(reader io.Reader, outputChan chan<- string, description string, silent bool, spin *spinner.Spinner) {
 	result := bytes.Buffer{}
 	scanner := bufio.NewScanner(reader)
 	for scanner.Scan() {
 		result.Write([]byte(scanner.Text()))
 		if !silent {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Stop()
+			}
 			klog.V(1).Infof("%s | %s", description, scanner.Text())
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Start()
+			}
```

Comment:
In terms of code changes this is the only thing I'm not sure about, calling to `Stop()` and `Start()` here when we are sharing a single `spin` and we have always two outputs that will be "fighting" at some extent for this resource.

Reply:
We need to stop spinner before write something in terminal.
Without stop and start, spinner marks will be remained in terminal. 
[==>   ] I1016 14:19:22.989369   19390 states.go:35] === applying state kubeadm.reset === 
So the outputs can be either terminal or file output. If a user redirects the outputs to file, spinner will not start.
I am not sure if I get your question correctly.

Reply:
It depends on start and stop implementation. From my understanding, it does take care.